gorm/ConnectionFactory_postgres.go

```diff
@@ -23,32 +24,36 @@ func NewPostgresConnectionFactory(parentDsn Dsn) *PostgresConnectionFactory {
 }
 
 func (f *PostgresConnectionFactory) CreateDbConnection(ctx context.Context, dbName string, repo DbMigraterSeeder) (*gorm.DB, *errors.GenericError) {
+	logger := log.WithContext(ctx)
 	// Connect to parent DB
 	dbName = strings.ReplaceAll(dbName, "-", "_") // Postgres doesn't like dashes
 	parentDb, err := gorm.Open(postgres.Open((f.parentDsn.String())))
 	defer func() {
 		if rawDB, err := parentDb.DB(); err != nil {
+			logger.WithError(err).Errorf("unable to get raw connection for %s: %v\n", f.parentDsn, err)
+		} else {
 			if err = rawDB.Close(); err != nil {
-				log.Printf("unable to close connection for %s: %v\n", f.parentDsn, err)
+				logger.WithError(err).Errorf("unable to c\"test-db-connection\"lose connection for %s: %v\n", f.parentDsn, err)
```

Comment:
Instead of escaping the quotes, please use a raw string instead (also note typo fixes and `\n` removal):
```suggestion
				logger.WithError(err).Errorf(`unable to "test-db-connection" lose connection for %s: %v`, f.parentDsn, err)
```